src/Yesod/Csp.hs

```diff
@@ -36,12 +41,29 @@ import           Yesod.Core
 -- >     [whamlet|hello|]
 --
 cspPolicy :: (MonadHandler m) => DirectiveList -> m ()
-cspPolicy = addHeader "Content-Security-Policy" . directiveListToHeader
+cspPolicy = addHeader cspHeaderName . directiveListToHeader
+
+cspHeaderName :: Text
+cspHeaderName = "Content-Security-Policy"
 
 -- | Returns a generated Content-Security-Policy header.
 getCspPolicy :: DirectiveList -> Text
 getCspPolicy = directiveListToHeader
 
+-- | Creates a WAI 'Middleware' to add a Content-Security-Policy
+-- header to every response.
+cspMiddleware :: DirectiveList -> Middleware
+cspMiddleware = addHeaderMiddleware . mkHeader . directiveListToHeader
+  where
+    addHeaderMiddleware = modifyResponse . mapResponseHeaders . insertAt 5
```

Comment:
Sorry for the delay here. Can you explain why 5?

Reply:
Oops, replied by email and it went into the PR instead of here. So that it won't be O(n) on number of headers, but also won't put this as the first header. 5 = arbitrary small constant.